components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -10517,49 +10517,105 @@ private void addAPIPolicy(APIPolicy policy, Connection conn) throws SQLException
     /**
      * Update a API level throttling policy to database.
      * <p>
-     * policy will be inserted with that policy Id.
+     * If condition group already exists for the policy, that condition Group will be deleted and condition Group will
+     * be inserted to the database with old POLICY_ID.
      * </p>
      *
      * @param policy policy object defining the throttle policy
-     * @throws SQLException
+     * @throws APIManagementException
      */
-    private void updateAPIPolicy(APIPolicy policy, Connection conn) throws SQLException {
-        ResultSet resultSet = null;
-        PreparedStatement policyStatement = null;
-        String addQuery = SQLConstants.ThrottleSQLConstants.INSERT_API_POLICY_WITH_ID_SQL;
-        int policyId = policy.getPolicyId();
+    public APIPolicy updateAPIPolicy(APIPolicy policy) throws APIManagementException {
+        PreparedStatement updateStatement = null;
```

Comment:
move Prepared statements to the places using with try resource

Reply:
https://github.com/wso2/carbon-apimgt/pull/9211/commits/ac69c0fa8a7c202a9d45ac30e24258f8a757a95a